[Bug] [connector-cassandra] when field in source(cassancra) is timestamp type and data is null, will cause error.

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

source table in cassandra, when the field is timestamp type, and the data is null ,will cause nullpointer exception.

### SeaTunnel Version

2.3.3 and dev

### SeaTunnel Config

```conf
test1.config's content as below:


env {
    execution.parallelism = 1
    job.mode = "BATCH"
    checkpoint.interval = 10000
}
source {
    Cassandra {
        host = "192.168.0.100:9042"
        keyspace = "store"
        username = "admin"
        password = "123456"
        datacenter = "datacenter1"
        cql = "select userid,item_count,last_update_timestamp from shopping_cart"
    }
}

sink {
    doris {
        fenodes = "192.168.0.101:8030"
        username = "root"
        password = "123456"
        database = "cassandradb"
        table = "shopping_cart"
        table.identifier = "cassandradb.shopping_cart"
        sink.label-prefix = "seatunnel_offline"
        sink.enable-2pc = "false"
        doris.config = {
            format = "json",
            read_json_by_line = "true"
        }
        sink.properties.columns = "userid,item_count,last_update_timestamp"
    }
}
```

### Running Command

```shell
bin/seatunnel.sh --config test1.confg -e local
```

### Error Exception

```log
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:191)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: java.lang.NullPointerException
        at java.util.Objects.requireNonNull(Objects.java:203)
        at org.apache.seatunnel.connectors.seatunnel.cassandra.util.TypeConvertUtil.buildSeaTunnelRow(TypeConvertUtil.java:185)
        at org.apache.seatunnel.connectors.seatunnel.cassandra.source.CassandraSourceReader.lambda$pollNext$0(CassandraSourceReader.java:73)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.seatunnel.connectors.seatunnel.cassandra.source.CassandraSourceReader.pollNext(CassandraSourceReader.java:73)
        at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:150)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:95)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:168)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:100)
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:613)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)

        at org.apache.seatunnel.engine.client.job.ClientJobProxy.waitForJobComplete(ClientJobProxy.java:122)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:184)
        ... 2 more
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/util/TypeConvertUtil.java
+++ b/seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/util/TypeConvertUtil.java
@@ -44,6 +44,7 @@ import java.net.InetAddress;
 import java.net.UnknownHostException;
 import java.nio.ByteBuffer;
 import java.sql.Timestamp;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
@@ -181,9 +182,8 @@ public class TypeConvertUtil {
                     fields[i] = row.getLocalDate(i);
                     break;
                 case ProtocolConstants.DataType.TIMESTAMP:
-                    fields[i] =
-                            Timestamp.from(Objects.requireNonNull(row.getInstant(i)))
-                                    .toLocalDateTime();
+                    Instant instant = row.getInstant(i);
+                    fields[i] = instant == null ? null : Timestamp.from(instant).toLocalDateTime();
                     break;
                 case ProtocolConstants.DataType.BLOB:
                     fields[i] =

--- a/seatunnel-connectors-v2/connector-cassandra/src/test/java/org/apache/seatunnel/connectors/seatunnel/cassandra/util/TypeConvertUtilTest.java
+++ b/seatunnel-connectors-v2/connector-cassandra/src/test/java/org/apache/seatunnel/connectors/seatunnel/cassandra/util/TypeConvertUtilTest.java
@@ -1,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.cassandra.util;
+
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.datastax.oss.driver.api.core.cql.ColumnDefinition;
+import com.datastax.oss.driver.api.core.cql.ColumnDefinitions;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.type.DataTypes;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TypeConvertUtilTest {
+
+    @Test
+    void testBuildSeaTunnelRowKeepsNullTimestamp() {
+        Row row = mock(Row.class);
+        ColumnDefinitions columnDefinitions = mock(ColumnDefinitions.class);
+        ColumnDefinition columnDefinition = mock(ColumnDefinition.class);
+
+        when(row.size()).thenReturn(1);
+        when(row.getColumnDefinitions()).thenReturn(columnDefinitions);
+        when(columnDefinitions.get(0)).thenReturn(columnDefinition);
+        when(columnDefinition.getType()).thenReturn(DataTypes.TIMESTAMP);
+        when(row.getInstant(0)).thenReturn(null);
+
+        SeaTunnelRow seaTunnelRow = TypeConvertUtil.buildSeaTunnelRow(row);
+
+        Assertions.assertNull(seaTunnelRow.getField(0));
+    }
+}

